Handling Presentational role conflict resolution (#1189)
act-rules/act-rules.github.io: pages/glossary/hidden.md

---
title: Explicitly Hidden
key: hidden
unambiguous: true
objective: true
input_aspects:
  - CSS styling
  - DOM tree
---

> An HTML element is _explicitly hidden_ if at least one of the following is true for itself or any of its ancestors:

Reviewer: ```suggestion
An HTML element is _hidden_ if at least one of the following is true for itself or any of its [ancestors][] in the [flat tree][]:
```

- has a `hidden` attribute; or
- has a CSS property `display` set to `none`; or

> - has a CSS property `visibility` set to `hidden`.

Reviewer 2: ```suggestion
- has a [computed](https://www.w3.org/TR/css-cascade/#computed-value) CSS property `display` of `none`; or
- has a [computed](https://www.w3.org/TR/css-cascade/#computed-value) CSS property `visibility` of `hidden`.
```

> - has a [computed](https://www.w3.org/TR/css-cascade/#computed-value) CSS property `visibility` of `hidden`.

Reviewer: ```suggestion
- has a [computed](https://www.w3.org/TR/css-cascade/#computed-value) CSS property `visibility` of `hidden`.
- has an `aria-hidden` attribute set to `true`
```

For semantic role to work it needs aria-hidden too.

---

> title: Explicitly Hidden

Reviewer: Thinking about this a little more... I don't think I like having a "hidden" definition that isn't the opposite of the "visible" definition. We did that in axe-core and it's pretty confusing. I think we should rename this to "hidden state", and say it can be either true or false.

Author: Which is why I called it "explicitly hidden" in the first place https://github.com/act-rules/act-rules.github.io/pull/1189#discussion_r397726136 😆 
"hidden state" is also a good name.
